spec/unit/lib/cloud_controller/metrics/periodic_updater_spec.rb

```diff
@@ -353,6 +368,124 @@ module VCAP::CloudController::Metrics
+            queue: VCAP::CloudController::Jobs::Queues.local(VCAP::CloudController::Config.config),
+            run_at: Time.now
+          )
+          periodic_updater.update_job_queue_load
+
+          expected_pending_job_queue_load = {
+            'cc-api-0': 1
+          }
+          expected_total = 1
+
+          expect(statsd_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load, expected_total)
+          expect(prometheus_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load)
+        end
+      end
+
+      context 'when local queue does not have pending jobs' do
+        it 'emits the local queue load as 0 for discoverability' do
+          periodic_updater.update_job_queue_load
+
+          expected_pending_job_queue_load = {
+            'cc-api-0': 0
+          }
+          expected_total = 0
+
+          expect(statsd_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load, expected_total)
+          expect(prometheus_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load)
+        end
+      end
+
+      it 'includes the load of the delayed job queue and the total' do
+        Delayed::Job.enqueue(VCAP::CloudController::Jobs::Runtime::EventsCleanup.new(1), queue: 'cc_local', run_at: Time.now)
+        Delayed::Job.enqueue(VCAP::CloudController::Jobs::Runtime::EventsCleanup.new(1), queue: 'cc_local', run_at: Time.now)
+        Delayed::Job.enqueue(VCAP::CloudController::Jobs::Runtime::EventsCleanup.new(1), queue: 'cc_generic', run_at: Time.now)
+
+        periodic_updater.update_job_queue_load
+
+        expected_pending_job_queue_load = {
+          cc_local: 2,
+          cc_generic: 1,
+          'cc-api-0': 0
+        }
+        expected_total = 3
+
+        expect(statsd_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load, expected_total)
+        expect(prometheus_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load)
+      end
+
+      it 'finds jobs which have not been attempted yet' do
+        Delayed::Job.enqueue(VCAP::CloudController::Jobs::Runtime::EventsCleanup.new(1), queue: 'cc_local', run_at: Time.now)
+        Delayed::Job.enqueue(VCAP::CloudController::Jobs::Runtime::EventsCleanup.new(1), queue: 'cc_generic', run_at: Time.now)
+
+        periodic_updater.update_job_queue_load
+
+        expected_pending_job_queue_load = {
+          cc_local: 1,
+          cc_generic: 1,
+          'cc-api-0': 0
+        }
+        expected_total = 2
+
+        expect(statsd_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load, expected_total)
+        expect(prometheus_updater).to have_received(:update_job_queue_load).with(expected_pending_job_queue_load)
+      end
+
+      it 'ignores jobs that have already been attempted' do
```

Comment:
Did you check whether there are any jobs which attempt's are increased and they get rerun?

Reply:
I've checked, and what you're talking about is possible, but we have yet to realize it

Reply:
Conclusion: instead of using `attempts`, using `failed_at` to filter all jobs that are not failed before